File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int minRow = 0;
        int maxCol = matrix[0].size() - 1;
        
        //cout  << "Rows = " << rows << ", Columns = " << cols << endl;
        while ( minRow < matrix.size() && maxCol >= 0 )
        {
            if (target == matrix[minRow][maxCol])
                return true;
            else if (target < matrix[minRow][maxCol])
                maxCol--;
            else 
                minRow++;
                
        }    
        
        return false;
    }
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        minRow = 0
        maxCol = len(matrix[0]) - 1
        
        while minRow < len(matrix) and maxCol >= 0:
            if matrix[minRow][maxCol] == target:
                return True
            elif target < matrix[minRow][maxCol]:
                maxCol -= 1
            else:
                minRow += 1
        
        return False